test/files/run/t12405.check

```diff
@@ -24,7 +24,7 @@ package <empty> {
     final <synthetic> def equals$extension[A]($this: C[A])(x$1: Any): Boolean = {
   case <synthetic> val x1: Any = x$1;
   case5(){
-    if (x1.isInstanceOf[C[A]])
+    if (x1.isInstanceOf[C[$this._]])
```

Comment:
Looks wrong, no?

Reply:
I think it's right, the synthetic `equals` method used to match `case _: C[A]` where `C[A]` was a weird ill-formed TypeRef in which the argument symbol `A` is class C's type parameter symbol. Now the synthetic `equals` method becomes `case _: C[_]`, where `_` references new AbstractTypeSymbol (this is the change in SyntheticMethods), same as in a hand-written match.

Maybe `_`'s owner is wrong though (`$this`), not the same as what the typer would generate. But I don't think that matters.